### DISCUSSION TITLE
MEPLHAA Script please help

### DISCUSSION THREAD

**Discussion by aptyp222** [RavenSystem/esp-homekit-devices#2003] [MEPLHAA Scripts]
Hello, I love your project. Please help to adapt to the new firmware 12.0.2. Before that, I used the site for automatic configuration https://glumb.github.io/haa-configurator/ I hope sooner or later they will adapt to the new system. I have two devices

1.
{"c":{"l":13,"b":[{"g":0,"t":5}]},"a":[{"0":{"r":[{"g":5,"v":0}]},"1":{"r":[{"g":5,"v":1}]},"t":20,"b":[{"g":0,"t":1}],"s":0,"w":3,"d":3600},{"t":7,"s":0,"f0":[{"g":12,"t":1,"i":1}],"f1":[{"g":12,"i":1}]}]}

2.
{"c":{"o":1},"a":[{"t":5,"s":0,"f1":[{"g":4,"i":0}],"f0":[{"g":4,"t":0,"i":0}]},{"t":7,"s":0,"f0":[{"g":14,"i":1,"t":0}],"f1":[{"g":14,"i":1}]},{"t":7,"s":0,"f0":[{"g":13,"t":0,"i":1}],"f1":[{"g":13,"i":1}]},{"t":7,"s":0,"f0":[{"g":12,"t":0,"i":1}],"f1":[{"g":12,"i":1}]}]}
 p.s
Do you have any plans to develop a similar site for the configuration???

**Comment by xrust83** [comment]
1.
{"c":{"io":[[[13],3],[[0,12],6,1],[[5],2]],"l":13,"b":[{"g":0,"t":5}]},"a":[{"0":{"r":[{"g":5,"v":0}]},"1":{"r":[{"g":5,"v":1}]},"t":20,"b":[{"g":0,"t":1}],"s":0,"w":3,"d":3600},{"t":7,"s":0,"f0":[{"g":12,"t":1,"i":1}],"f1":[{"g":12,"i":1}]}]}

2.
{"c":{"io":[[[4,14,13,12],6,1]],"o":1},"a":[{"t":5,"s":0,"f1":[{"g":4,"i":0}],"f0":[{"g":4,"t":0,"i":0}]},{"t":7,"s":0,"f0":[{"g":14,"i":1,"t":0}],"f1":[{"g":14,"i":1}]},{"t":7,"s":0,"f0":[{"g":13,"t":0,"i":1}],"f1":[{"g":13,"i":1}]},{"t":7,"s":0,"f0":[{"g":12,"t":0,"i":1}],"f1":[{"g":12,"i":1}]}]}

I don't know if you have external pull-up resistors. Try this

**Comment by arcclaro** [comment]
Unfortunately I have the same issue as aptyp222.

I can't understand fully the new requirements for input/output definitions...

my current script is:

{"c":{"l":13,"b":[{"g":0,"t":5}]},"a":[{"0":{"r":[{"g":12}]},"1":{"r":[{"g":12,"v":1}]},"b":[{"g":0}],"s":0},{"0":{"r":[{"g":5}]},"1":{"r":[{"g":5,"v":1}]},"b":[{"g":9}],"s":0}]}

can anyone give me some help and convert it to the new standard?


One more thing, is tehere any planned update to the https://glumb.github.io/haa-configurator/ , it would be the perfect companion to HAA!

**Reply by xrust83** [reply]
{"c":{"io":[[[13],3],[[12,5],2],[[0,9],6]],"l":13,"b":[{"g":0,"t":5}]},"a":[{"0":{"r":[{"g":12}]},"1":{"r":[{"g":12,"v":1}]},"b":[{"g":0}],"s":0},{"0":{"r":[{"g":5}]},"1":{"r":[{"g":5,"v":1}]},"b":[{"g":9}],"s":0}]}

**Comment by aptyp222** [comment]
@xrust83 Thanks, you helped me a lot. As for the leak sensor, they are now showing activity. How can you change them? They are connected directly to the payment without resistors. ( ground and gpio )
{"c":{"io":[[[4,14,13,12],6,1]],"o":1},"a":[{"t":5,"s":0,"f1":[{"g":4,"i":0}],"f0":[{"g":4,"t":0,"i":0}]},{"t":7,"s":0,"f0":[{"g":14,"i":1,"t":0}],"f1":[{"g":14,"i":1}]},{"t":7,"s":0,"f0":[{"g":13,"t":0,"i":1}],"f1":[{"g":13,"i":1}]},{"t":7,"s":0,"f0":[{"g":12,"t":0,"i":1}],"f1":[{"g":12,"i":1}]}]}

**Reply by xrust83** [reply]
try inverting gpio, 6.1.1 or 6.0.1. but you can still see the diagram.

{"c":{"io":[[[4,14,13,12],6,0,1]],"o":1},"a":[{"t":5,"s":0,"f1":[{"g":4,"i":0}],"f0":[{"g":4,"t":0,"i":0}]},{"t":7,"s":0,"f0":[{"g":14,"i":1,"t":0}],"f1":[{"g":14,"i":1}]},{"t":7,"s":0,"f0":[{"g":13,"t":0,"i":1}],"f1":[{"g":13,"i":1}]},{"t":7,"s":0,"f0":[{"g":12,"t":0,"i":1}],"f1":[{"g":12,"i":1}]}]}

**Reply by xrust83** [reply]
{"c":{"io":[[[14,13,12],6,0,1],[[4],6]],"o":1},"a":[{"t":5,"s":0,"f1":[{"g":4}],"f0":[{"g":4,"t":0}]},{"t":7,"s":0,"f0":[{"g":14,"t":0}],"f1":[{"g":14}]},{"t":7,"s":0,"f0":[{"g":13,"t":0}],"f1":[{"g":13}]},{"t":7,"s":0,"f0":[{"g":12,"t":0}],"f1":[{"g":12}]}]}

Hello. I have reviewed your jsion more carefully try now

**Comment by wrx5** [comment]
Can I also get new code for my light switches please?

I have single button and double button switches with below config

Single button: 
-----------------
{"c": {"n": "Single Light Switch","l": 13,"i": 0,"b": [{"g": 0,"t": 5}]},"a": [{"0": {"r": [{"g": 12,"v": 0}]},"1": {"r": [{"g": 12,"v": 1}]},"t": 1,"b": [{"g": 0,"t": 1}],"s": 0,"f0": [],"f1": []},{"i": 0.5,"1": {"s": [{"a": 1}]}}]}

Double button:
------------------
{"c":{"n": "Double Switch","l": 2,"i": 0,"b":[{"g":5,"t":5}]},"a":[{"0":{"r":[{"g":13,"v":0}]},"1":{"r":[{"g":13,"v":1}]},"t":1,"b":[{"g":16,"t":1}],"s":5,"f0":[],"f1":[]},{"0":{"r":[{"g":12,"v":0}]},"1":{"r":[{"g":12,"v":1}]},"t":1,"b":[{"g":5,"t":1}],"s":5},{"i":0.5,"1":{"s":[{"a":1}]}}]}

**Reply by xrust83** [reply]
{"c": {"io":[[[13],3],[[0],6],[[12],2]],"n": "Single Light Switch","l": 13,"i": 0,"b": [{"g": 0,"t": 5}]},"a": [{"0": {"r": [{"g": 12,"v": 0}]},"1": {"r": [{"g": 12,"v": 1}]},"t": 1,"b": [{"g": 0,"t": 1}],"s": 0,"f0": [],"f1": []},{"i": 0.5,"1": {"s": [{"a": 1}]}}]}

**Reply by xrust83** [reply]
{"c":{"io":[[[5,16],6,1],[[12,13],2],[[2],3]],"n": "Double Switch","l": 2,"i": 0,"b":[{"g":5,"t":5}]},"a":[{"0":{"r":[{"g":13,"v":0}]},"1":{"r":[{"g":13,"v":1}]},"t":1,"b":[{"g":16,"t":1}],"s":5,"f0":[],"f1":[]},{"0":{"r":[{"g":12,"v":0}]},"1":{"r":[{"g":12,"v":1}]},"t":1,"b":[{"g":5,"t":1}],"s":5},{"i":0.5,"1":{"s":[{"a":1}]}}]}

**Reply by rafsew74** [reply]
Hello, I love your project. Please help to adapt to the new firmware 12.0.3.
{"c":{"l":13,"b":[{"g":0,"t":5}],"i":1},"a":[{"0":{"r":[{"g":12,"v":1,"i":0.5}]},"1":{"a":0},"2":{"r":[{"g":12,"i":0.5},{"g":12,"v":1,"i":0.5},{"g":12,"v":1,"i":2}]},"3":{"a":2},"t":40,"d":10,"f3":[{"g":3,"t":0}],"f4":[{"g":3}],"s":0,"b":[]},{"5":{"r":[]},"t":22,"b":[],"g":1,"n":3,"j":10}]}

**Reply by rafsew74** [reply]
2.
{"c":{"l":13,"b":[{"g":0,"t":5}]},"a":[{"t":40,"d":18,"f3":[{"g":14,"t":0}],"f4":[{"g":14}],"f2":[{"g":4,"t":0}],"f5":[{"g":4}],"0":{"r":[{"g":12,"v":1,"i":0.5}]},"1":{"a":0},"2":{"r":[{"g":12,"i":0.5},{"g":12,"v":1,"i":0.5},{"g":12,"v":1,"i":2}]},"3":{"a":2},"10":{"a":0}}]}

**Reply by xrust83** [reply]
> {"c":{"l":13,"b":[{"g":0,"t":5}],"i":1},"a":[{"0":{"r":[{"g":12,"v":1,"i":0.5}]},"1":{"a":0},"2":{"r":[{"g":12,"i":0.5},{"g":12,"v":1,"i":0.5},{"g":12,"v":1,"i":2}]},"3":{"a":2},"t":40,"d":10,"f3":[{"g":3,"t":0}],"f4":[{"g":3}],"s":0,"b":[]},{"5":{"r":[]},"t":22,"b":[],"g":1,"n":3,"j":10}]}

{"c":{"io":[[[13,12],2],[[0],6],[[3],6,1]],"l":13,"b":[{"g":0,"t":5}],"i":1},"a":[{"0":{"r":[{"g":12,"v":1,"i":0.5}]},"1":{"a":0},"2":{"r":[{"g":12,"i":0.5},{"g":12,"v":1,"i":0.5},{"g":12,"v":1,"i":2}]},"3":{"a":2},"t":40,"d":10,"f3":[{"g":3,"t":0}],"f4":[{"g":3}],"s":0,"b":[]},{"5":{"r":[]},"t":22,"b":[],"g":1,"n":3,"j":10}]}

**Reply by xrust83** [reply]
> {"c":{"l":13,"b":[{"g":0,"t":5}]},"a":[{"t":40,"d":18,"f3":[{"g":14,"t":0}],"f4":[{"g":14}],"f2":[{"g":4,"t":0}],"f5":[{"g":4}],"0":{"r":[{"g":12,"v":1,"i":0.5}]},"1":{"a":0},"2":{"r":[{"g":12,"i":0.5},{"g":12,"v":1,"i":0.5},{"g":12,"v":1,"i":2}]},"3":{"a":2},"10":{"a":0}}]}

{"c":{"io":[[[13,12],2],[[0],6],[[4,14],6,1]],"l":13,"b":[{"g":0,"t":5}]},"a":[{"t":40,"d":18,"f3":[{"g":14,"t":0}],"f4":[{"g":14}],"f2":[{"g":4,"t":0}],"f5":[{"g":4}],"0":{"r":[{"g":12,"v":1,"i":0.5}]},"1":{"a":0},"2":{"r":[{"g":12,"i":0.5},{"g":12,"v":1,"i":0.5},{"g":12,"v":1,"i":2}]},"3":{"a":2},"10":{"a":0}}]}

**Reply by rafsew74** [reply]
{"c":{"l":13,"b":[{"g":0,"t":5}],"i":1},"a":[{"0":{ "r":[{"g":12,"v":1,"i":0.5}]},"1":{"a":0},"2":{"r":[{ "g":12,"i":0.5},{"g":12,"v":1,"i":0.5},{"g":12,"v":1,"i": 2}]},"3":{"a":2},"t":40,"d":10,"f3":[{"g":3,"t":0}]," f4":[{"g":3}],"s":0,"b":[]},{"5":{"r":[]},"t":22,"b" :[],"g":1,"n":3,"j":10}]}

{"c":{"io":[[[13,12],2],[[0],6],[[3],6,1]],"l":13,"b": [{"g":0,"t":5}],"i":1},"a":[{"0":{"r":[{"g":12,"v": 1,"i":0.5}]},"1":{"a":0},"2":{"r":[{"g":12,"i":0.5},{"g ":12,"v":1,"i":0.5},{"g":12,"v":1,"i":2}]},"3":{"a":2} ","t":40,"d":10,"f3":[{"g":3,"t":0}],"f4":[{"g":3}],"s" :0,"b":[]},{"5":{"r":[]},"t":22,"b":[],"g":1,"n":3, "j":10}]}
doesn't work, device is not responding

**Reply by xrust83** [reply]
What does gpio3 do in the schematic?

**Reply by rafsew74** [reply]
Gate with one opening and closing sensor and TE sensormperatury

**Reply by xrust83** [reply]
This should be correct https://github.com/RavenSystem/esp-homekit-devices/wiki/Garage-Door#garage-door-example

**Reply by xrust83** [reply]
{"c":{"io":[[[13,12],2],[[0],6],[[3],6,1]],"l":13,"b":[{"g":0,"t":5}]},"a":[{"0":{ "r":[{"g":12,"v":1,"i":0.5}]},"1":{"a":0},"2":{"r":[{ "g":12,"i":0.5},{"g":12,"v":1,"i":0.5},{"g":12,"v":1,"i": 2}]},"3":{"a":2},"t":40,"d":10,"f3":[{"g":3,"t":0}]," f4":[{"g":3}],"s":0,"b":[]},{"5":{"r":[]},"t":22,"b" :[],"g":1,"n":3,"j":10}]}

**Reply by wrx5** [reply]
> {"c":{"io":[[[5,16],6,1],[[12,13],2],[[2],3]],"n": "Double Switch","l": 2,"i": 0,"b":[{"g":5,"t":5}]},"a":[{"0":{"r":[{"g":13,"v":0}]},"1":{"r":[{"g":13,"v":1}]},"t":1,"b":[{"g":16,"t":1}],"s":5,"f0":[],"f1":[]},{"0":{"r":[{"g":12,"v":0}]},"1":{"r":[{"g":12,"v":1}]},"t":1,"b":[{"g":5,"t":1}],"s":5},{"i":0.5,"1":{"s":[{"a":1}]}}]}


Single switch is working fine. Thanks for that. However double switch doesn't work properly. First button on the switch doesn't work until I press the second button. I have to press second button to get first button working.

**Comment by rafsew74** [comment]
doesn't work, Hello, the temperature sensor is connected to GPIO1, and the opening sensor to GPIO3.

**Reply by rafsew74** [reply]
Thanks for the help! I Solved the problem !

**Comment by JonnyFPV** [comment]
I also don't understand the new version :)

Please can anyone help me? It's a simple relay board to open an electric door opener.

So the relay needs to be triggered for 5 Seconds and then the relay need to turn off automatically.

Old configuration was working good:

`{
  "c": {
    "l": 13,
    "b": [
      {
        "g": 0,
        "t": 5
      }
    ]
  },
  "a": [
    {
      "0": {
        "r": [
          {
            "g": 12,
            "v": 1,
            "i": 5
          }
        ]
      },
      "1": {
        "r": [
          {
            "g": 13,
            "v": 0
          }
        ]
      },
      "t": 4,
      "b": [],
      "s": 1,
      "i": 5
    }
  ]
}
`

Thanks :-)

**Reply by JonnyFPV** [reply]
so if anyone looking for exact this. I found a solution. 

Situation: single relay board with 1 relay connected to gpio 12 and a button on gpio5.
 it's used for an electric door unlock mechanism. inching time is set to 4 seconds. 

`{
  "c":{ "io":[[[12],2],[[5],6,0,1]] },
  "a":[{
    "t":4,
    "i":4,
    "0":{"r":[[12,1,4]]},
    "1":{"r":[[12]]},
    "b":[[5,0]]
  }]
}`

**Comment by spiseregern** [comment]
Hi
I have upgraded my whole house today, I have a range of 1,2,3 and 4 gang light switches ,i used to merlin converting program everything when fine except button 1 on all my light switches no longer works unless i push any of the buttons on the switch, unfortunatley i can not get my single gang switches to work as they only have one button, they all still work fine in HomeKit

this is my merlin meplhaa script for my single gang switches, any help would be apprecated

{"c":{"o":0},"a":[{"0":{"r":[{"g":13,"v":0}]},"1":{"r":[{"g":13,"v":1}]},"t":1,"b":[{"g":16,"t":1,"i":1}],"s":0}]}

**Reply by xrust83** [reply]
> Hi
> 
> I have upgraded my whole house today, I have a range of 1,2,3 and 4 gang light switches ,i used to merlin converting program everything when fine except button 1 on all my light switches no longer works unless i push any of the buttons on the switch, unfortunatley i can not get my single gang switches to work as they only have one button, they all still work fine in HomeKit
> 
> 
> 
> this is my merlin meplhaa script for my single gang switches, any help would be apprecated
> 
> 
> 
> {"c":{"o":0},"a":[{"0":{"r":[{"g":13,"v":0}]},"1":{"r":[{"g":13,"v":1}]},"t":1,"b":[{"g":16,"t":1,"i":1}],"s":0}]}

https://github.com/RavenSystem/esp-homekit-devices/discussions/2034#discussioncomment-5958687

**Reply by spiseregern** [reply]
sorry that was my old melphaa script this is the one for merlin with single gang button that wont work

{"c":{"o":0,"io":[[[13],2],[[16],6]]},"a":[{"0":{"r":[[13,0]]},"1":{"r":[[13,1]]},"t":1,"b":[[16,1]],"s":0}]}

**Reply by spiseregern** [reply]
thanks mate that helped me, all fixed now cheers

**Comment by childledj** [comment]
Hi everyone!
I have troubled with sonoff 4ch r3 - nothing working when I've change it's script
Peregrine working was {"c":{"l":13,"i":1},"a":[{"0":{"r":[{"g":12,"v":0}]},"1":{"r":[{"g":12,"v":1}]},"b":[{"g":0 ,"p":0},{"g":0,"t":1}]},{"0":{"r":[{"g":5, "v":0}]},"1":{"r":[{"g":5 ,"v":1}]},"b":[{"g":9 ,"p":0},{"g": 9,"t":1}]},
{"0":{"r":[{"g":4, "v":0}]},"1":{"r":[{"g":4 ,"v":1}]},"b":[{"g":10,"p":0},{"g":10,"t":1}]},
{"0":{"r":[{"g":15,"v":0}]},"1":{"r":[{"g":15,"v":1}]},"b":[{"g":14,"p":0},{"g":14,"t":1}]}]}

I've tried to replace first {"l":13,"i":1} with {"io":[[[13],3],1]} and with {"io":[[[13],3,1]]} - but without success

Thanks for helping here!

And it will be amazing if someone could make Peregrine2Merlin script converter (if it is possible)

**Reply by spiseregern** [reply]
https://converter-to-merlin.netlify.app/

**Reply by childledj** [reply]
Awesome!!! Thanks a lot!

**Comment by antropoperso** [comment]
I need some help to. After the update, most of my devices are not responding, I see them in the home app and in HAA, but no response to actions. I am trying to tweak and replace the scripts but with mediocre success.
The most important gate - garage door (sonoff) (with two sensors for fully open & close state)

Previous script that works fine with v11:
{"c":{"l":13,"b":[{"g":0,"t":5}]},"a":[{"0":{"r":[{"g":12,"v":1,"i":0.5}]},"1":{"a":0},"2":{"r":[{"g":12,"i":0.5},{"g":12,"v":1,"i":0.5},{"g":12,"v":1,"i":2}]},"3":{"a":2},"10":{"a":0},"t":40,"d":18,"f3":[{"g":14,"t":0}],"f4":[{"g":14}],"f2":[{"g":4,"t":0}],"f5":[{"g":4}],"s":0,"b":[{"g":0,"t":0,"i":0}]}]}

My attempt to run the same functionality in v12:
{"c":{"l":13,"b":[[0,5]],"io":[[[13],3],[[0,14,4],6],[[12],2]]},"a":[{"0":{"r":[[12,1]]},"1":{"a":0},"2":{"r":[[12],[12,1],[12,1]]},"3":{"a":2},"10":{"a":0},"t":40,"d":18,"f3":[[14,0]],"f4":[[14]],"f2":[[4,0]],"f5":[[4]],"s":0,"b":[[0,0]]}]}

and when i'm open door, they stay opened permanently. Only when I put device back into setup mode, it release the relay and I can close the gate using the switch on the wall...  I'm out of ideas

**EDIT**
i have working one - just in case if someone have similar issue:
{"c":{"io":[[[12,13],2],[[0],6],[[14],6,1]],"l":13,"b":[[0,5]]},"a":[{"t":40,"d":18,"0":{"r":[[12,1,0.5]]},"1":{"a":0},"2":{"r":[[12,0,1.5],[12,1,0.5],[12,1,2]]},"3":{"a":2},"f3":[[14,0]],"f4":[[14]]}]}
